cli/pa_analyze_command.go

```diff
@@ -0,0 +1,1155 @@
+					if lastSeq > highestLastSeq {
+						highestLastSeq = lastSeq
+						highestLastSeqServer = serverName
+					}
+				}
+				cmd.logDebug(
+					"Stream %s / %s highest last sequence: %d @ %s",
+					accountName,
+					streamName,
+					highestLastSeq,
+					highestLastSeqServer,
+				)
+
+				// Check if some server's sequence is below warning threshold
+				maxDelta := uint64(float64(highestLastSeq) * lastSequenceLagThreshold)
+				threshold := uint64(0)
+				if maxDelta <= highestLastSeq {
+					threshold = highestLastSeq - maxDelta
+				}
+				for serverName, streamDetail := range replicasStreamDetails {
+					lastSeq := streamDetail.State.LastSeq
+					if lastSeq < threshold {
+						examples.Addf(
+							"%s/%s server %s lastSequence: %d is behind highest lastSequence: %d on server: %s",
+							accountName,
+							streamName,
+							serverName,
+							lastSeq,
+							highestLastSeq,
+							highestLastSeqServer,
+						)
+						laggingReplicas += 1
+					}
+				}
+			}
+		}
+	}
+
+	cmd.logInfo("Inspected %d streams across %d accounts", len(streamsInspected), len(accountsWithStreams))
+
+	if laggingReplicas > 0 {
+		cmd.logIssue("Found %d replicas lagging behind", laggingReplicas)
+		cmd.logExamples(examples)
+		return SomeIssues, nil
+	}
+	return Pass, nil
+}
+
+// checkCpuUsage verify that CPU usage is below a certain threshold for all known servers
+func (cmd *paAnalyzeCmd) checkCpuUsage(r *archive.Reader) (paCheckOutcome, error) {
+	const (
+		cpuThreshold = 0.9 // Warn using >90% CPU
+	)
+
+	examples := newCollectionOfExamples(cmd.examplesLimit)
+
+	severVarsTag := archive.TagServerVars()
+
+	for _, clusterName := range r.GetClusterNames() {
+		clusterTag := archive.TagCluster(clusterName)
+		for _, serverName := range r.GetClusterServerNames(clusterName) {
+			serverTag := archive.TagServer(serverName)
+			var serverVarz server.Varz
+			if err := r.Load(&serverVarz, serverTag, clusterTag, severVarsTag); errors.Is(err, archive.ErrNoMatches) {
+				cmd.logWarning("Artifact 'VARZ' is missing for server %s", serverName)
+				continue
+			} else if err != nil {
+				return Skipped, fmt.Errorf("failed to load VARZ for server %s: %w", serverName, err)
+			}
+
+			// Example: 350% usage with 4 cores => 87.5% averaged
+			averageCpuUtilization := serverVarz.CPU / float64(serverVarz.Cores)
+
+			if averageCpuUtilization > cpuThreshold {
+				examples.Addf("%s - %s: %.1f%%", clusterName, serverName, averageCpuUtilization)
+			}
+		}
+	}
+
+	if examples.Count() > 0 {
+		cmd.logIssue("Found servers with high CPU usage")
+		cmd.logExamples(examples)
+		return SomeIssues, nil
+	}
+
+	return Pass, nil
+}
+
+// checkHighSubjectCardinalityStreams verify that the number of unique subjects is below some magic number for each known stream
+func (cmd *paAnalyzeCmd) checkHighSubjectCardinalityStreams(r *archive.Reader) (paCheckOutcome, error) {
+	const (
+		numSubjectsThreshold = 1_000_000 // Warn if a stream has >1M unique subjects
```

Comment:
You might have noticed some checks have magic numbers hardcoded. We're going with recommended values from support folks for now. Would be nice to have some sort of profile/configuration later on.